pkg/odo/cli/component/devfile.go

```diff
@@ -62,7 +62,7 @@ func (po *PushOptions) DevfilePush() (err error) {
 		platformContext = nil
 	} else {
 		kc := kubernetes.KubernetesContext{
-			Namespace: po.namespace,
+			Namespace: po.KClient.Namespace,
```

Comment:
I had wondered that myself. I think it came up in one of the merges I did but couldn't see where it actually came from. 
Looking at the real odo master branch it has "Namespace: po.KClient.Namespace," so I think it may have come up when I did a rebase against master (or tried to).